### ISSUE TITLE
E0606 not detected in a simple loop

### ISSUE THREAD

**Issue by hydrargyrum** [pylint-dev/pylint#9642]
### Bug description

```python
"""doc"""

def f():
    """doc"""

    for i in []:
        if i:
            fail1 = 42
    print(fail1)  # bug detected

    for i in []:
        fail2 = 42
    print(fail2)  # bug not detected
```
```


### Configuration

_No response_

### Command used

```shell
`pylint bad.py`
```

### Pylint output

```python
************* Module bad
bad.py:9:10: E0606: Possibly using variable 'fail1' before assignment (possibly-used-before-assignment)

------------------------------------------------------------------
Your code has been rated at 3.75/10 (previous run: 3.75/10, +0.00)
```
```


### Expected behavior

line 13 should also have `possibly-used-before-assignment` for `fail2`

### Pylint version

```shell
pylint 3.2.0
astroid 3.2.1
Python 3.11.9
```
